File: internal/content/testdata/bench/typescript/nc-reconstruct-itinerary.ts
function find_itinerary(tickets: string[][]): string[] {
    const graph: {[key: string]: string[]} = {};
    // Sort tickets in reverse so that pop() yields lexicographically smallest
    tickets.sort((a, b) => {
        if (a[0] !== b[0]) return b[0] < a[0] ? -1 : 1;
        return b[1] < a[1] ? -1 : 1;
    });
    for (const [src, dst] of tickets) {
        if (!(src in graph)) graph[src] = [];
        graph[src].push(dst);
    }

    const result: string[] = [];

    function dfs(airport: string): void {
        while (graph[airport] && graph[airport].length > 0) {
            dfs(graph[airport].pop()!);
        }
        result.push(airport);
    }

    dfs("JFK");
    return result.reverse();
}

File: internal/content/testdata/bench/csharp/nc-reconstruct-itinerary.cs
public class Solution {
    public string[] find_itinerary(string[][] tickets) {
        var graph = new Dictionary<string, List<string>>();
        // Sort tickets in reverse so that RemoveLast() yields lexicographically smallest
        var sortedTickets = tickets.OrderBy(t => t[0]).ThenBy(t => t[1]).Reverse().ToArray();
        foreach (var ticket in sortedTickets) {
            if (!graph.ContainsKey(ticket[0])) graph[ticket[0]] = new List<string>();
            graph[ticket[0]].Add(ticket[1]);
        }
        var result = new List<string>();
        Dfs("JFK", graph, result);
        result.Reverse();
        return result.ToArray();
    }

    private void Dfs(string airport, Dictionary<string, List<string>> graph, List<string> result) {
        var neighbors = graph.ContainsKey(airport) ? graph[airport] : new List<string>();
        while (neighbors.Count > 0) {
            string next = neighbors[neighbors.Count - 1];
            neighbors.RemoveAt(neighbors.Count - 1);
            Dfs(next, graph, result);
        }
        result.Add(airport);
    }
}
